src/psyclone/tests/psyir/frontend/fparser2_derived_type_test.py

```diff
@@ -278,14 +330,28 @@ def test_derived_type_ref(f2008_parser):
     assert isinstance(amem, ArrayOfStructuresMember)
     assert isinstance(amem.member, ArrayMember)
     assert isinstance(amem.member.indices[0], Range)
-    assign = assignments[4]
+    assert isinstance(amem.member.indices[0].children[0], BinaryOperation)
+    assert amem.member.indices[0].children[0].children[0].symbol.name == "var"
+    gen = fortran_writer(amem)
+    assert (gen == "subgrid(3)%data(:)")
     # var%region%subgrid(3)%data(var%start:var%stop)
+    assign = assignments[4]
     amem = assign.lhs.member.member.member
     assert isinstance(amem, ArrayMember)
     assert isinstance(amem.indices[0], Range)
     assert isinstance(amem.indices[0].children[0], StructureReference)
     assert isinstance(amem.indices[0].children[0].member, Member)
     assert amem.indices[0].children[0].member.name == "start"
+    # vars(1)%region%subgrid(3)%data(:) = 1.0
+    assign = assignments[5]
+    amem = assign.lhs
+    gen = fortran_writer(amem)
+    assert (gen == "vars(1)%region%subgrid(3)%data(:)")
+    # vars(:)%region%subgrid(3)%xstop
+    assign = assignments[6]
+    amem = assign.lhs
+    gen = fortran_writer(amem)
+    assert (gen == "vars(:)%region%subgrid(3)%xstop")
```

Comment:
pylint says the parenthesis are not needed

Reply:
Done.